File: extractor-go/internal/domain/I18n.go
package domain

type I18n struct {
	PreviousHistoryID NullableInt64
	HistoryID         NullableInt64
	I18nId            string
	FileVersion       int32
	ContainerFile     string
	EnText            string
	PtBrText          string
	Active            bool
	Deleted           bool
}

func (i *I18n) Equals(otherI18n I18n) bool {
	return (i.I18nId == otherI18n.I18nId &&
		i.ContainerFile == otherI18n.ContainerFile &&
		i.EnText == otherI18n.EnText &&
		i.PtBrText == otherI18n.PtBrText &&
		i.Active == otherI18n.Active)
}

type MinI18n struct {
	I18nId     string
	LastUpdate string
	PtBrText   string
}

type I18nValue struct {
	I18nId   string
	PtBrText string
}

File: ui/src/models/I18n.ts
export type I18n = {
	PreviousHistoryID: number | null;
	HistoryID: number | null;
	I18nId: number;
	FileVersion: number;
	ContainerFile: string;
	EnText: string;
	PtBrText: string;
	Active: boolean;
	Deleted: boolean;
};

export type MinI18n = {
	I18nId: number;
	LastUpdate: string;
	PtBrText: string;
};
